src/codes.ts

```diff
@@ -8,38 +8,33 @@ class ErrorInvalidCode extends Error {
   }
 }
 
-enum UnspentType {
-  p2sh = 'p2sh',
-  p2shP2wsh = 'p2shP2wsh',
-  p2wsh = 'p2wsh',
-}
+enum EUnspentType { p2sh, p2shP2wsh, p2wsh }
 
-enum Purpose {
-  internal = 'internal',
-  external = 'external',
-}
+const UnspentType = tcomb.enums(EUnspentType, 'UnspentType');
```

Comment:
this also doesn't do what we want
```
 > tcomb1.enums(X)
{ [Function: Enums]
  meta: 
   { kind: 'enums',
     map: { '0': 'a', '1': 'b', '2': 'c', a: 0, b: 1, c: 2 },
     name: undefined,
     identity: true },
  displayName: '"0" | "1" | "2" | "a" | "b" | "c"',
  is: [Function] }

> tcomb1.enums(X)('x')
/home/bitgotto/src/BitGo/indexerdb-microservice/typescript/unspentselection/node_modules/tcomb/lib/fail.js:2
  throw new TypeError('[tcomb] ' + message);
  ^

TypeError: [tcomb] Invalid value "x" supplied to "0" | "1" | "2" | "a" | "b" | "c" (expected one of [
  "0",
  "1",
  "2",
  "a",
  "b",
  "c"
])

```

Reply:
Is there a way to run a typescript shell? I was trying to test this stuff out and I couldn't run a typescript shell so I was copy pasting the dist files, which must have caused this because I thought this worked.

Reply:
https://github.com/TypeStrong/ts-node